Fix cutscene in COP 2-5 "Ancient Vows"

MarianArlt opened the pull request:
<!-- place 'x' mark between square [] brackets to affirm: -->
**_I affirm:_**
- [x] that I agree to Project Topaz's [Limited Contributor License Agreement](http://project-topaz.com/blob/release/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

_ Fixes a wrong cutscene ( #1420  ) after entering through the Dilapidated Gate in Misareaux Coast (F-7) and clicking the Spacial Displacement
_ This affected mission 4-2 (csid 8) so I added one optional dialogue and a new default dialogue there while at it
_ Minor style guide clean up for the Dilapidated Gate (4383c20098a31dcbfd507bd758f228c0568b1c97)

Captures used for this PR:
[Missions: COP 2-4 (An Eternal Melody) ~ COP 2-5 (Ancient Vows)](https://www.youtube.com/watch?v=xHDoAEIQ7_Y&feature=youtu.be)
[Chains of Promathia: COP 4-1 (Sheltering Doubt)](https://www.youtube.com/watch?v=TyOlnk3c_Po&feature=youtu.be)
[Chains of Promathia: COP 4-2 (The Savage)](https://www.youtube.com/watch?v=jIVHFjG1kS0&feature=youtu.be)

MarianArlt committed:
Style guide clean up

MarianArlt committed:
Fix COP 2-5 cutscene

MarianArlt committed:
optional dialogue

zach2good commented on scripts/zones/Misareaux_Coast/npcs/Spatial_Displacement.lua:
```diff
@@ -11,10 +12,12 @@ end
+    if player:getCurrentMission(COP) == tpz.mission.id.cop.THE_SAVAGE and player:getCharVar("PromathiaStatus") == 0 then
         player:startEvent(8)
+    -- COP 4-1
+    elseif (player:hasCompletedMission(COP, tpz.mission.id.cop.SHELTERING_DOUBT)) then
```
Please remove these outer brackets

MarianArlt replied:
Missed that one :)

zach2good commented on scripts/zones/Tavnazian_Safehold/npcs/Despachiaire.lua:
```diff
@@ -5,6 +5,7 @@
 -----------------------------------
 require("scripts/globals/missions")
 -----------------------------------
+-- TO DO: Quest giver for "X Marks the Spot" which requires COP 2-5 Ancient Vows completed
```
We tend to search on `TODO` when we leave things like this in the code, so please change this to that. 

I think this is also a good opportunity to make an issue and put the issue number in with the TODO:
```
If you agree with staff and/or your reviewers that some work in your pull request can
 be left as "to-do", make new issues on GitHub for your new TODO items and put the 
ID alongside the comment. The comment should also be sufficiently descriptive of what
 the missing work is, and why it was left out. eg. // TODO: A Boy's Dream - PLD AF Quest 2 - 
Cannot be completed until fishing is implemented (GitHub Issue #12345)
```

MarianArlt replied:
Done. See #1481
I added all missing quests and the issue reference.

zach2good reviewed (changes requested):
Looks fine to me, thanks for all the cleanup!

Just two small style dings, then this is ready to go.

MarianArlt commented:
Please check Despechiaire again, I cleaned that one up too, structured it a little to make it legible and added comments.
The "Sheltering Doubt" optional dialogue needed a missionStatus condition or else it would trigger after going to Bastok.
I moved the event references to the bottom; wouldn't want to delete those, somebody put in effort to pick them, but they shouldn't clutter the functions.

(Force Push fixed a missing Unix (LF) line end)

MarianArlt committed:
Style Guide Clean Up

ibm2431 reviewed (commented):
Was initially merged into a safekeeping branch, but I'll "approve" this and merge the branch into release.
